### PR TITLE
feat: #357 http-server OIDC /login endpoint

### LINKED ISSUE #357: OIDC /login endpoint
part of #353 

Need to implement an endpoint from which the UI can fetch the OIDC authN/Z request.

### PR DESCRIPTION
**Pull request by llorllale**
~~depends on #359~~ 

closes #357 

Signed-off-by: George Aristy <george.aristy@securekey.com>

### CODE REVIEW

File: pkg/restapi/oidc/dependencies.go [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+	return v.v.Verify(ctx, token)
+}
+
+func (o *oidcProviderImpl) Endpoint() oauth2.Endpoint {
+	return o.op.Endpoint()
+}
+
+type idToken interface {
+	Claims(interface{}) error
+}
```
Comment (Baha-sk) [inline pkg/restapi/oidc/dependencies.go:46]: seems odd to have to create an interface that is only used in returns of this file, I would prefer to use OIDC's interface directly instead but can't seem to find one, is this intentional?

Comment (llorllale) [reply pkg/restapi/oidc/dependencies.go:46]: The intent is to be able to mock these dependencies in unit tests in order to have good coverage.

**Review by Baha-sk — APPROVED** [review]

File: pkg/restapi/common/common.go [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+// WriteErrorResponsef write error resp.
+func WriteErrorResponsef(rw http.ResponseWriter, logger log.Logger, status int, msg string, args ...interface{}) {
+	rw.WriteHeader(status)
+
+	err := json.NewEncoder(rw).Encode(ErrorResponse{
+		Message: fmt.Sprintf(msg, args...),
+	})
+
+	if err != nil {
+		logger.Errorf("Unable to send error message, %s", err)
```
Comment (DRK3) [inline pkg/restapi/common/common.go:38]: Just for consistency, consider using a `:` instead of a `,`

**Review by DRK3 — APPROVED** [review]
Left minor nitpick (optional)

File: pkg/restapi/common/common.go (hunk 2 above)
Comment (llorllale) [reply pkg/restapi/common/common.go]: done

**Pull request merged**